packages/plugin-browserify/src/plugin.ts

```diff
@@ -1,9 +1,70 @@
 import { Transform, TransformCallback } from 'stream';
+import { promises as fs } from 'fs';
+import os from 'os';
+import pathUtils from 'path';
 import { BrowserifyObject } from 'browserify';
-import { postProcessBundle, PostProcessOptions } from '@metamask/snap-utils';
+import {
+  checkManifest,
+  evalBundle,
+  postProcessBundle,
+  PostProcessOptions,
+} from '@metamask/snap-utils';
 import { fromSource } from 'convert-source-map';
 
-export type Options = PostProcessOptions;
+const TEMP_BUNDLE_PATH = pathUtils.join(os.tmpdir(), 'snaps-bundle.js');
+
+type PluginOptions = {
+  eval?: boolean;
+  manifestPath?: string;
+  writeManifest?: boolean;
+};
+
+export type Options = PluginOptions &
+  Omit<PostProcessOptions, 'sourceMap' | 'inputSourceMap'>;
+
+/**
+ * Run eval on the processed bundle and fix the manifest, if configured.
+ *
+ * @param options - The plugin options.
+ * @param code - The code to eval, if the eval option is enabled.
+ */
+async function postBundle(options: Partial<Options>, code: string) {
+  if (options.eval) {
+    await fs.mkdir(pathUtils.dirname(TEMP_BUNDLE_PATH), { recursive: true });
+    await fs.writeFile(TEMP_BUNDLE_PATH, code);
+
+    await evalBundle(TEMP_BUNDLE_PATH)
+      .catch((error) => {
+        throw new Error(`Snap evaluation error: ${error.toString()}`);
+      })
+      .finally(() => fs.unlink(TEMP_BUNDLE_PATH));
+  }
+
+  if (options.manifestPath) {
+    const { errors, warnings } = await checkManifest(
+      pathUtils.dirname(options.manifestPath),
+      options.writeManifest,
+      code,
+    );
+
+    if (!options.writeManifest && errors.length > 0) {
+      console.error(`Manifest Error: The manifest is invalid.`);
+      errors.forEach((error) => console.error(`Manifest Error: ${error}`));
+
+      process.exit(1);
```

Comment:
Is there a reason not to throw here? We do that for eval failures

Reply:
This is copied from the CLI, but I can change it to throw instead.